### ISSUE TITLE
Pass `publish` argument to preview hook

### ISSUE THREAD

**Issue by ghing** [tarbell-project/tarbell#369]
The [calls the preview hook](https://github.com/tarbell-project/tarbell/blob/1.0.4/tarbell/app.py#L519), passing only the site instance.  It would be nice to pass the publish argument to this hook to allow for separate behaviors depending if a page is being previewed during local development or being baked for deployment.

In general, I feel like many of Tarbell's hooks would benefit from being passed the arguments that are passed to the methods where the hooks are invoked.

**Comment by eads** [comment]
Yes, you're right. Some of those can be implicitly determined by looking at
the state of the site object, but that's not a very nice pattern.

Let's queue this up for 1.1

On Tue, Sep 8, 2015 at 11:33 AM, Geoffrey Hing <notifications@github.com>
wrote:

> The calls the preview hook
> <https://github.com/tarbell-project/tarbell/blob/1.0.4/tarbell/app.py#L519>,
> passing only the site instance. It would be nice to pass the publish
> argument to this hook to allow for separate behaviors depending if a page
> is being previewed during local development or being baked for deployment.
>
> In general, I feel like many of Tarbell's hooks would benefit from being
> passed the arguments that are passed to the methods where the hooks are
> invoked.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/tarbell-project/tarbell/issues/369>.
>



-- 
David Eads | http://recoveredfactory.net

"Medical statistics will be our standard of measurement: we will weigh life
for life and see where the dead lie thicker, among the workers or among the
privileged." -- Rudolf Virchow

**Comment by ghing** [comment]
@eads, off the top of your head is there a way to determine publish vs. non-publish from attributes of the site object?

**Comment by eads** [comment]
Off the top of my head using something like `publish =
bool(site.app.config.get('BUILD_PATH'))` should do the trick in the current
version.

On Tue, Sep 8, 2015 at 3:13 PM, Geoffrey Hing <notifications@github.com>
wrote:

> @eads <https://github.com/eads>, off the top of your head is there a way
> to determine publish vs. non-publish from attributes of the site object?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/tarbell-project/tarbell/issues/369#issuecomment-138673583>
> .
>



-- 
David Eads | http://recoveredfactory.net

"Medical statistics will be our standard of measurement: we will weigh life
for life and see where the dead lie thicker, among the workers or among the
privileged." -- Rudolf Virchow
